`assets/` folder now includes the filename root, to avoid collisions …

roblabs opened the pull request:
…with existing folders

-----

Thanks for the package!!   This is my **most** favorite package of Atom.io.  I added this small tweak, as I had several markdown files in the same directory and I needed to package up `.md` and `assets` separately. 

👍

roblabs committed:
`assets/` folder now includes the filename root, to avoid collisions with existing folders

tlnagy commented:
Thanks for the PR. I would prefer to offer this as an optional setting. Could you add it as an option in the config section: https://github.com/tlnagy/atom-markdown-image-assistant/blob/aea8edbc15d2e6c934c38affe166c540bfc3e0a4/lib/main.coffee#L8

roblabs committed:
Supports 3 options for `assets` folder — 
1) Default,
2) explicit name, or
3) name derived from the root name of `.md` filename

roblabs commented:
My apologies for not seeing / understanding the config options.  I redid my original PR to rework the logic to support three cases, in this hierarchy.  That is, explicitly changing the folder name overrides all other settings for folder name.

1) Explicitly changing folder name in Settings, e.g., `the-dogs-assets/`
2) Use folder name derived from the Markdown file name, e.g., `README-assets/`
3) Default folder name `assets/`



All cases have been tested.

Again, your package is awesome.  Thanks again.

tlnagy commented on README.md:
```diff
@@ -4,9 +4,15 @@ Add sane drag and drop and copy/paste support for images to markdown
+![](README.assets/README-54317719.png)
+
+-----
 
```
This description makes this look like the default. Maybe add a section called options and add this description under there? 

Also, I would be careful about replacing `assets/` with `<fileroot>.assets/`. The former does not have the dot.

tlnagy commented on lib/main.coffee:
```diff
@@ -18,11 +20,16 @@ module.exports = MarkdownImageAssistant =
             type: 'boolean'
             default: false
+        preserveFileNameInAssetsFolder:
+            title: "Use the Markdown filename when creating the assets folder"
```
Maybe say, "Create per-file asset directories."?

Also, maybe include a description why this might be helpful: "This can help organize assets and prevent name collisions."

tlnagy commented on lib/main.coffee:
```diff
@@ -80,7 +87,10 @@ module.exports = MarkdownImageAssistant =
 
-        assets_dir = path.basename(atom.config.get('markdown-image-assistant.imageDir'))
+        if atom.config.get('markdown-image-assistant.imageDir') == defaultImageDir && atom.config.get('markdown-image-assistant.preserveFileNameInAssetsFolder')
+            assets_dir = path.basename(path.parse(target_file).name + "." + atom.config.get('markdown-image-assistant.imageDir'))
```
Nice use of `path.parse`! I didn't know about that function. 

```

tlnagy reviewed (changes requested):
Just a couple changes (mainly wording).

roblabs committed:
support for markup.  Update README with settings.

roblabs commented:
If the README is not exactly, what you want, then I suggest pulling the functionality and then editing the Readme.  Again, great Atom package.  I use this everyday, and with the new functionality, even more!

tlnagy committed:
Update wording

tlnagy commented:
Thanks! All your changes have been merged. I'll push out a new version of the package here in a sec.

roblabs commented:
Great!  Also, thanks for reviewing my README wording.  One of your suggestions did not make it (inadvertently), so thanks for catching that.
